test/distributed/elastic/test_control_plane.py

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env python3
+# Owner(s): ["oncall: distributed"]
+
+import os
+import pickle
+import socket
+import tempfile
+from contextlib import contextmanager
+
+import requests
+from requests.adapters import HTTPAdapter
+
+from torch.distributed.elastic.control_plane import main, TORCH_WORKER_SERVER_SOCKET
+from torch.testing._internal.common_utils import requires_cuda, run_tests, TestCase
+from urllib3.connection import HTTPConnection
+from urllib3.connectionpool import HTTPConnectionPool
+
+
+class UnixHTTPConnection(HTTPConnection):
+    def __init__(self, socket_path: str) -> None:
+        super().__init__("localhost")
+
+        self.socket_path = socket_path
+
+    def connect(self) -> None:
+        self.sock = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
+        self.sock.connect(self.socket_path)
+
+
+class UnixHTTPConnectionPool(HTTPConnectionPool):
+    def __init__(self, socket_path: str) -> None:
+        super().__init__("localhost")
+
+        self.socket_path = socket_path
+
+    def _new_conn(self):
+        return UnixHTTPConnection(self.socket_path)
+
+
+class UnixHTTPAdapter(HTTPAdapter):
+    def __init__(self, socket_path: str) -> None:
+        super().__init__()
+
+        self.socket_path = socket_path
+
+    def get_connection(self, url, proxies=None):
+        return UnixHTTPConnectionPool(self.socket_path)
+
+    def request_url(self, request, proxies):
+        print(request.path_url)
+        return request.path_url
+
+
+@contextmanager
+def local_worker_server() -> None:
+    with tempfile.TemporaryDirectory() as tmpdir:
+        socket_path = os.path.join(tmpdir, "socket.sock")
+        os.environ[TORCH_WORKER_SERVER_SOCKET] = socket_path
+
+        print("starting")
+        with main():
+            session = requests.Session()
+            session.mount("http+unix://", UnixHTTPAdapter(socket_path))
+            yield session
+
+
+class WorkerServerTest(TestCase):
+    def test_worker_server(self) -> None:
+        with local_worker_server() as session:
+            resp = session.get("http+unix://worker/")
+            self.assertEqual(resp.status_code, 200)
+            self.assertEqual(resp.text, "torch.distributed.WorkerServer")
+
+            resp = session.post("http+unix://worker/handler/ping")
+            self.assertEqual(resp.status_code, 200)
+            self.assertEqual(resp.text, "pong")
+
+    @requires_cuda
+    def test_dump_nccl_trace_pickle(self) -> None:
+        with local_worker_server() as session:
+            resp = session.post("http+unix://worker/handler/dump_nccl_trace_pickle")
```

Comment:
UX:
- Do we need to register the handler first? 
- If not for some cases, how would users know what is enabled by default?

Reply:
I wonder if we should adopt some RESTful API approach?

Reply:
handlers do need to be registered before the post request -- we're currently statically registering all handlers so they'll always be present by this call

Reply:
RESTful is a pretty overloaded term could you clarify what you mean? I'm not really sure how to express "dump_nccl_trace_pickle" in the rest model since it doesn't really map to a data object with CRUD

Reply:
i guess naively, `GET "/worker/handler/nccl_trace"` seems 'restful' as you're reading the state of an object.  I'm not sure if the RESTful model accounts for an object that is constantly changing, but otherwise that seems ok to me?

or is the difference that we are requesting a dump to disk in this call vs a return of the current value?

Reply:
This returns the data directly. For this one I think it's more or less fine to do get, but for some other potential handlers it's weirder. "Start tracing for 10 seconds" is not a very restful behavior. 

I think the prior art for that is a CREATE/POST call to create the profiling and then other requests to get the data. i.e.

```
CREATE /profile {"time": 10} -> {"id": 15}
GET /profile/15 -> {...}
```

vs

```
POST /handler/profile {"time": 10} -> {...}
```

Reply:
> RESTful is a pretty overloaded term could you clarify what you mean?

Worth exploring [HATEOAS](https://en.wikipedia.org/wiki/HATEOAS) that could support various cases given pluggable nature of handlers.

Reply:
If we were going that route I think something like openapi or Swagger might be best. Could potentially generate thrift defs off of Swagger as well